pkg/components/prometheus-operator/component.go

```diff
@@ -162,6 +164,29 @@ func (c *component) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContex
 		c.Prometheus.Ingress.SetDefaults()
 	}
 
+	// If user has provided `ingress` block in prometheus and `externalURL` both then they should be
+	// same.
+	if c.Prometheus != nil && c.Prometheus.ExternalURL != "" && c.Prometheus.Ingress != nil {
+		exu, err := url.Parse(c.Prometheus.ExternalURL)
+		if err != nil {
+			return hcl.Diagnostics{
+				{
+					Severity: hcl.DiagError,
+					Summary:  fmt.Sprintf("unable to parse 'external_url' with value %q as URL: %v", c.Prometheus.ExternalURL, err),
+				},
+			}
+		}
+
+		if exu.Host != c.Prometheus.Ingress.Host {
+			return hcl.Diagnostics{
+				{
+					Severity: hcl.DiagError,
+					Summary:  "'external_url' and 'prometheus.ingress.host' does not match",
```

Comment:
```suggestion
					Summary:  "'external_url' and 'prometheus.ingress.host' do not match",
```

- Fix grammar.
- Shouldn't we say `prometheus.external_url` if we specify the full config "path" of the other knob?